fuse_models/include/fuse_models/common/sensor_proc.h

```diff
@@ -465,7 +461,197 @@ inline bool processDifferentialPoseWithCovariance(
+    //
+    //        (R1 | 0)   (cos(yaw1) -sin(yaw1) 0)
+    // J_p2 = (------) = (sin(yaw1)  cos(yaw1) 0)
+    //        ( 0 | 1)   (        0          0 1)
+    //
+    //
+    //
+    // Therefore, for the the covariance propagation of [1] we would get the following jacobians:
+    //
+    //        (1 0 | -sin(yaw1) * x12 - cos(yaw1) * y12)
+    // J_p1 = (0 1 |  cos(yaw1) * x12 - sin(yaw1) * y12)
+    //        (0 0 |                                  1)
+    //
+    //         (R1 | 0)   (cos(yaw1) -sin(yaw1) 0)
+    // J_p12 = (------) = (sin(yaw1)  cos(yaw1) 0)
+    //         ( 0 | 1)   (        0          0 1)
+    //
+    //
+    //
+    // At this point we could go one step further since p12 = t12 * dt and include the jacobian of this additional
+    // equation:
+    //
+    // J_t12 = dt * Id
+    //
+    // where Id is a 3x3 identity matrix.
+    //
+    // However, that would give us the covariance of the twist t12, and here we simply need the one of the relative
+    // pose p12.
+    //
+    //
+    //
+    // Finally, since we need the inverse of the jacobian J_p12, we can use the inverse directly:
+    //
+    //            ( cos(yaw1) sin(yaw1) 0)
+    // J_p12^-1 = (-sin(yaw1) cos(yaw1) 0)
+    //            (         0         0 1)
+    //
+    //
+    //
+    // In the implementation below we use:
+    //
+    // sy = sin(-yaw1)
+    // cy = cos(-yaw1)
+    //
+    // which are defined before.
+    //
+    // Therefore, the jacobians end up with the following expressions:
+    //
+    //        (1 0 | sin(-yaw1) * x12 - cos(-yaw1) * y12)
+    // J_p1 = (0 1 | cos(-yaw1) * x12 + sin(-yaw1) * y12)
+    //        (0 0 |                                   1)
+    //
+    //            (cos(-yaw1) -sin(-yaw1) 0)
+    // J_p12^-1 = (sin(-yaw1)  cos(-yaw1) 0)
+    //            (         0           0 1)
+    //
+    //
+    //
+    // Note that the covariance propagation expression derived here for dependent pose measurements gives more accurate
+    // results than simply changing the sign in the expression for independent pose measurements, which would be:
+    //
+    // C12 = J_p2 * C2 * J_p2^T - J_p1 * C1 * J_p1^T
+    //
+    // where J_p1 and J_p2 are the jacobians for p12 = p1^-1 * p2 (we're abusing the notation here):
+    //
+    //        (-cos(-yaw1),  sin(-yaw1),  sin(-yaw1) * x12 + cos(-yaw1) * y12)
+    // J_p1 = (-sin(-yaw1), -cos(-yaw1), -cos(-yaw1) * x12 + sin(-yaw1) * y12)
+    //        (          0,           0,                                   -1)
+    //
+    //        (R1 | 0)   (cos(yaw1) -sin(yaw1) 0)
+    // J_p2 = (------) = (sin(yaw1)  cos(yaw1) 0)
+    //        ( 0 | 1)   (        0          0 1)
+    //
+    // which are the j_pose1 and j_pose2 jacobians used above for the covariance propagation expresion for independent
+    // pose measurements.
+    //
+    // This seems to be the approach adviced in https://github.com/cra-ros-pkg/robot_localization/issues/356, but after
```

Comment:
The motivation of this PR is the same as https://github.com/cra-ros-pkg/robot_localization/issues/356